### PR TITLE
feat(22307): journey step and connections

### PR DESCRIPTION
**Pull request by nnorbert**
Journey steps and connections API

### CODE REVIEW

**Review by peschina — COMMENTED** [review]
Looks good, let's just update the permissions

File: apps/integrator-api/src/routes/journeys/index.ts [hunk 1]
```diff
@@ -2,29 +2,29 @@ import { FastifyInstance } from "fastify";
...
+  ParamsWithJourneyId,
+  UpdateJourneyBodyDO,
 } from "../../plugins/entities/journey/types";
 
 const TAGS = ["Journeys"];
 
 export default async function journeys(app: FastifyInstance) {
   app.get<{
-    Reply: GenericResponse<JourneysDO> | Error;
+    Reply: GenericResponse<JourneysDO[]> | Error;
```
Comment (peschina) [inline apps/integrator-api/src/routes/journeys/index.ts:27]: `JourneysDO` is already an array

File: apps/integrator-api/src/plugins/entities/journey/index.ts [hunk 2]
```diff
@@ -6,20 +6,20 @@ import {
...
   JourneyStatusType,
-} from "../../../routes/schemas";
+} from "./types";
 
 export type JourneyPlugin = Awaited<ReturnType<typeof buildPlugin>>;
 
 const buildGetJourneys =
   (repo: JourneyRepo, log: FastifyBaseLogger) =>
-  async (organizationId: string): Promise<Journeys> => {
+  async (organizationId: string): Promise<JourneysDO[]> => {
```
Comment (peschina) [inline apps/integrator-api/src/plugins/entities/journey/index.ts:22]: `JourneysDO` is already an array

File: apps/integrator-api/src/plugins/entities/journey/types.ts [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
 } from "../../../routes/schemas";
 
 export type JourneyStatusType = Static<typeof JourneyStatus>;
 export type JourneyDetailsDO = Static<typeof JourneyDetails>;
 export type JourneyPublicDetailsDO = Static<typeof JourneyPublicDetails>;
 export type JourneysDO = Static<typeof Journeys>;
 
+export type FullJourneyDO = Static<typeof FullJourney>;
+export type CreateJourneyBodyDO = Static<typeof CreateJourneyBody>;
+export type UpdateJourneyBodyDO = Static<typeof UpdateJourneyBody>;
```
Comment (peschina) [inline apps/integrator-api/src/plugins/entities/journey/types.ts:20]: Do we want to keep this here even if it's not used in repo, but in route only? Same for ParamsWithJourneyId

**Review by peschina — COMMENTED** [review]
Almost there, just one type to fix I think

**Review by peschina — APPROVED** [review]

File: apps/integrator-api/src/routes/stepConnections/index.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  ParamsWithJourneyStepConnectionIdDO,
+} from "../schemas";
+import { formatAPIResponse } from "../../utils/responseFormatter";
+import { authPermissions } from "../../types/authPermissions";
+import {
+  CreateJourneyStepConnectionDO,
+  JourneyStepConnectionDO,
+} from "../../plugins/entities/journeyStepConnections/types";
+
+const TAGS = ["StepConnections"];
```
Comment (alfonsograziano) [inline apps/integrator-api/src/routes/stepConnections/index.ts:17]: Non blocking: At some point, we might want to export the tags in a different file (like `openApiUtils` and then import it, all good for now though

**Review by alfonsograziano — APPROVED** [review]
LGTM

**Pull request merged**